File: 2003/IntroToVBNETPart2/Presentation/Exception4.vb
Imports System

Public Class CentralErrorProcessing
    Shared Public Sub ProcessException ( ByVal e As Exception )
        Console.WriteLine ( ) 
        Console.WriteLine ( "HelpLink: {0}", e.HelpLink )
        Console.WriteLine ( "InnerException: {0}", e.InnerException )
        Console.WriteLine ( "Message: {0}", e.Message )
        Console.WriteLine ( "Source: {0}", e.Source )
        Console.WriteLine ( "StackTrace: {0}", e.StackTrace )
        Console.WriteLine ( "TargetSite: {0}", e.TargetSite )
    End Sub
End Class

Public Class Arithmetic
    
    Public Sub PerformOverflow ( )
        Try
            Dim value1 As Short = 10000
            Dim value2 As Short = 20000
            Dim result As Short 
            result = value1 * value2 
            Console.WriteLine ( "result: {0}", result ) 
        Catch e As Exception
            CentralErrorProcessing.ProcessException ( e ) 
        End Try
    End Sub
    Public Sub PerformDivideByZero ( )
        Try
            Dim zero As Integer = 0
            Dim result As Integer 
            result = CInt ( 123 / zero )
            Console.WriteLine ( "result: {0}", result )
        Catch e As Exception
            CentralErrorProcessing.ProcessException ( e ) 
        End Try
    End Sub
    
End Class

Public Class Exception4
    ' Another way to do Central Error Processing
    Shared Sub Main ( )
        Dim myArithmetic As New Arithmetic ( )
        myArithmetic.PerformDivideByZero ( )
        myArithmetic.PerformOverflow ( )
    End Sub
End Class

File: 2003/IntroToVBNETPart2/Presentation/Exception4.cs
using System ;

public class CentralErrorProcessing
{
    public static void ProcessException ( Exception e )
    {
        Console.WriteLine ( ) ;
        Console.WriteLine ( "HelpLink: {0}", e.HelpLink ) ;
        Console.WriteLine ( "InnerException: {0}", e.InnerException ) ;
        Console.WriteLine ( "Message: {0}", e.Message ) ;
        Console.WriteLine ( "Source: {0}", e.Source ) ;
        Console.WriteLine ( "StackTrace: {0}", e.StackTrace ) ;
        Console.WriteLine ( "TargetSite: {0}", e.TargetSite ) ;
    }
}

public class Arithmetic
{
    public void PerformOverflow ( )
    {
        try
        {
            short value1 = 10000 ;
            short value2 = 20000 ;
            short result ;
            result = (short) ( value1 * value2 ) ;
            Console.WriteLine ( "result: {0}", result ) ;
        }
        catch ( Exception e ) {
            CentralErrorProcessing.ProcessException ( e )  ;
        }
    }
    public void PerformDivideByZero ( )
    {
        try
        {
            int zero = 0 ;
            int result ;
            result = ( int ) ( 123 / zero ) ;
            Console.WriteLine ( "result: {0}", result ) ;
        }
        catch ( Exception e ) {
            CentralErrorProcessing.ProcessException ( e )  ;
        }
    }
}

public class Exception4
{
    // Another way to do Central Error Processing
    public static void Main ( )
    {
        Arithmetic myArithmetic = new Arithmetic ( ) ;
        myArithmetic.PerformDivideByZero ( ) ;
        myArithmetic.PerformOverflow ( ) ;
    }
}